src/accel/detail/HitProcessor.hh

```diff
@@ -32,48 +34,53 @@ namespace detail
  * Transfer Celeritas sensitive detector hits to Geant4.
  *
  * This serves a similar purpose to the \c G4FastSimHitMaker class for
- * generating hit objects. It should be "stream local" but by necessity has to
- * share the same HitManager which takes the input
- * detector names, maps them to detector IDs, and adds a HitCollector to the
- * action manager.
+ * generating hit objects. It \b must be thread-local because the sensitive
+ * detectors it stores are thread-local, and additionally Geant4 makes
+ * assumptions about object allocations that cause crashes if the HitProcessor
```

Comment:
A worker has its own event loops.  So, this is not an assumption  as the Geant4 MT is an event-level multithreading, but a consequence.

Reply:
The "assumption" is that objects like G4Navigator, G4Step, and G4Track can *only* be allocated by a worker thread, and that they're deallocated by the same worker thread. Those objects aren't necessarily associated with a specific event (in the G4 fast simulation some of those are reused across multiple events), nor are they inherently associated with a specific thread (except by the invisible implementation of the thread allocator).